src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs

```diff
@@ -678,86 +849,116 @@ public void RenderDrawables(VisualizationEventArgs e)
 
 #if DEBUG
             renderTimer.Start();
-#endif
-            Detach(false);
+#endif                  
+            Text = null;
 
             var packages = e.Packages.Concat(e.SelectedPackages)
                 .Cast<HelixRenderPackage>().Where(rp=>rp.MeshVertexCount % 3 == 0);
-
-            var points = HelixRenderPackage.InitPointGeometry();
-            var lines = HelixRenderPackage.InitLineGeometry();
-            var linesSel = HelixRenderPackage.InitLineGeometry();
-            var dynamoMesh = HelixRenderPackage.InitMeshGeometry();
-            var mesh = HelixRenderPackage.InitMeshGeometry();
+    
             var text = HelixRenderPackage.InitText3D();
 
             var aggParams = new PackageAggregationParams
             {
-                Packages = packages,
-                Points = points,
-                Lines = lines,
-                SelectedLines = linesSel,
-                DynamoMesh = dynamoMesh,
-                Mesh = mesh,
+                Packages = packages,                          
                 Text = text
             };
 
             AggregateRenderPackages(aggParams);
 
-            if (!points.Positions.Any())
-                points = null;
-
-            if (!lines.Positions.Any())
-                lines = null;
-
-            if (!linesSel.Positions.Any())
-                linesSel = null;
-
-            if (!text.TextInfo.Any())
-                text = null;
-
-            if (!dynamoMesh.Positions.Any())
-                dynamoMesh = null;
-
-            if (!mesh.Positions.Any())
-                mesh = null;
 
 #if DEBUG
             renderTimer.Stop();
             Debug.WriteLine(string.Format("RENDER: {0} ellapsed for compiling assets for rendering.", renderTimer.Elapsed));
             renderTimer.Reset();
             renderTimer.Start();
-#endif
-
-            var updateGraphicsParams = new GraphicsUpdateParams
+#endif        
+             
+            //Helix render the packages in certain order. Here, the BillBoardText has to be rendered
+            //after rendering all the geometry. Otherwise, the Text will not get rendered at the right 
+            //position. Also, BillBoardText gets attached only once. It is not removed from the tree everytime.
+            //Instead, only the geometry gets updated every time. Once it is attached (after the geometry), helix
+            // renders the text at the right position.
+            if (Text != null && Text.TextInfo.Any())
```

Comment:
@ramramps We should look at using `CollectionViewSource.SortDescriptions` to create a list of sort descriptions which handle this for us. I was able to do this for transparency sorting. You can add a custom `SortDescription` that will sort the collection based on a property and an order (ascending, descending). Not sure what the property we'd use here is but, the sort description will place objects without the requested property at the beginning of the list. So, maybe we use the `BillboardText3D.Text` property. Ping me and I'll show you how I used this with transparency.